src/app/containers/create-movie/MovieForm.jsx

```diff
@@ -0,0 +1,159 @@
+import { Formik, Form, Field } from "formik";
+import { genres } from "./movie-data";
+import { initialValues } from "./movie-data";
+import { movieValidationSchema } from "./validationSchema";
+
+const MovieForm = ({ mutate }) => {
+  return (
+    <Formik
+      initialValues={initialValues}
+      validationSchema={movieValidationSchema}
+      onSubmit={(values, { setSubmitting, resetForm }) => {
+        const movieData = {
+          title: values.title,
+          description: values.description,
+          genres: {
+            id: values.genres.id,
+            name: values.genres.name,
+          },
+          videoSource: values.videoSource,
+          cast: {
+            role: values.cast.role,
+            crew: {
+              id: values.cast.crew.id,
+              name: values.cast.crew.name,
+            },
+          },
+        };
+
+        mutate(movieData, {
+          onSuccess: () => {
+            resetForm();
+            setSubmitting(false);
+          },
+          onError: () => {
+            setSubmitting(false);
+          },
+        });
+      }}
+    >
+      {({ isSubmitting, setFieldValue, errors, touched }) => (
+        <div className="max-w-md mx-auto">
+          <Form className="flex flex-col space-y-4 items-center bg-gradient-to-tr from-gray-50 to-blue-100 shadow-md rounded-lg px-8 pt-6 pb-8 mb-4">
+            <Field
+              className="w-96 px-3 py-2 border border-gray-300 rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-500"
+              type="text"
+              name="title"
+              placeholder="Title"
+            />
+            {errors.title && touched.title && (
+              <div className="text-red-500 text-m mt-1">{errors.title}</div>
+            )}
+
+            <Field
+              className="w-96 px-3 py-2 border border-gray-300 rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-500"
+              type="text"
+              name="description"
+              placeholder="Description"
+            />
+            {errors.description && touched.description && (
+              <div className="text-red-500 text-m mt-1">
+                {errors.description}
+              </div>
+            )}
+
+            <Field
+              className="w-96 px-3 py-2 border border-white-300 rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-500"
+              as="select"
+              name="genres.id"
+              onChange={(e) => {
```

Comment:
It's a good practice for long logic like this to be extracted into a separate function